drivers: counter: fix mcux lptmr unit test issue.

Holt-Sun opened the pull request:
Fix the MCUX LPTMR counter driver alarm behavior so that, after an alarm
fires, the match value is reset to the maximum top value. Without this,
restarting the LPTMR may hit the previous match value almost
immediately, causing the counter_basic_api test
test_valid_function_without_alarm to fail intermittently.

Extend the counter_basic_api test suite with a dedicated test
configuration that enables CONFIG_COUNTER_MCUX_LPTMR_ALARM and runs
the same tests on any board that instantiates an nxp,lptmr device.
This verifies the alarm-only LPTMR configuration against the common
counter API.

Depends-On: https://github.com/zephyrproject-rtos/zephyr/pull/100558

Holt-Sun committed:
drivers: counter: fix mcux lptmr unit test issue.

The counter alarm irq shall set the match value
to the max top value after stopping the timer, otherwise
the next start of the lptmr counter may shortly match the
the previous match value thus cause test_valid_function_without_alarm
run failed.

Signed-off-by: Holt Sun <holt.sun@nxp.com>

ZhaoxiangJin commented:
are you fixing https://github.com/zephyrproject-rtos/zephyr/issues/100529? I just tried this patch, but alarm-related testcases still failed.

Holt-Sun commented:
> are you fixing #100529? I just tried this patch, but alarm-related testcases still failed.

You need to enable the COUNTER_MCUX_LPTMR_ALARM.

I am not directly aiming at https://github.com/zephyrproject-rtos/zephyr/issues/100529, I just realize the lptmr unittest is not actually run with latest added COUNTER_MCUX_LPTMR_ALARM. I am waiting your https://github.com/zephyrproject-rtos/zephyr/pull/100558 merged, then reenable the unit test for lptmr both alarm+top.

Holt-Sun committed:
tests: counter: add mcux LPTMR alarm mode coverage

Extend the counter_basic_api tests with a dedicated configuration that
enables the MCUX LPTMR driver in alarm mode.

The new testcase:
- Selects CONFIG_COUNTER_MCUX_LPTMR_ALARM
- Runs the existing counter_basic_api suite against nxp_lptmr devices
  when built for a board that instantiates the LPTMR

This verifies that the alarm-only LPTMR configuration passes the common
counter API tests without relying on the TOP callback feature.

Signed-off-by: Holt Sun <holt.sun@nxp.com>
